### PR TITLE
refactor: refactor indicator props

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | 1. Tabs discard `indicatorSize` prop and replaces with `indicator: { size: xxx }` <br /> 2. Tabs support `indicator: { align: xxx }` prop to set the alignment of the Tabs indicator <br /> 3. ConfigProvider discard `indicatorSize` prop and replaces it with `indicator: { size: xxx }`<br /> 4. ConfigProvider support `indicator: { align: xxx }` prop to set the Tabs indicator alignment |
| 🇨🇳 Chinese | 1. Tabs 组件废弃 `indicatorSize` 属性，改为 `indicator: { size: xxx }` 代替 <br /> 2. Tabs 组件支持 `indicator: { align: xxx }` 属性，设置 Tabs 指示条对齐方式 <br /> 3. ConfigProvider 组件废弃 `indicatorSize` 属性，改为 `indicator: { size: xxx }` 代替 <br /> 4. ConfigProvider 组件支持 `indicator: { align: xxx }` 属性，设置 Tabs 指示条对齐方式 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/tabs/index.en-US.md [hunk 1]
```diff
@@ -53,8 +53,8 @@ Common props ref：[Common props](/docs/react/common-props)
 | centered | Centers tabs | boolean | false | 4.4.0 |
 | defaultActiveKey | Initial active TabPane's key, if `activeKey` is not set | string | - |  |
 | hideAdd | Hide plus icon or not. Only works while `type="editable-card"` | boolean | false |  |
-| indicatorAlign | Customize align of indicator | `start` \| `center` \| `end` | `center` | 5.13.0 |
-| indicatorSize | Customize length of indicator, which is the same as tab by default | number \| (origin: number) => number | - | 5.9.0 |
+| indicator | Customize `length` and `align` of indicator, which is the same as tab by default | { size?: number \| (origin: number) => number; align: `start` \| `center` \| `end`; } | - | 5.13.0 |
+| indicatorSize (deprecated) | Customize length of indicator, which is the same as tab by default | number \| (origin: number) => number | - | 5.9.0 - 5.13.0 |
```
Comment (afc163) [inline components/tabs/index.en-US.md:57]: 是不是可以搞一个 deprecated 的标签样式出来。

File: components/tabs/index.tsx [hunk 2]
```diff
@@ -78,6 +80,12 @@ const Tabs: React.FC<TabsProps> & { TabPane: typeof TabPane } = (props) => {
       'breaking',
       '`onPrevClick` and `onNextClick` has been removed. Please use `onTabScroll` instead.',
     );
+
+    warning(
+      !(legacyIndicatorSize || tabs?.indicatorSize),
+      'deprecated',
+      '`indicatorSize` has deprecated. Please use `indicator={ size: ... }` instead.',
```
Comment (afc163) [inline components/tabs/index.tsx:87]:
```suggestion
      '`indicatorSize` has deprecated. Please use `indicator={{ size: ... }}` instead.',
```

**Comment by afc163** [comment]
Card 组件里也有一个 tabProps，看看需不需要改。

File: components/tabs/index.en-US.md (hunk 1 above)
Comment (li-jia-nan) [reply components/tabs/index.en-US.md:57]: 可以在 table 单元格里面加个 tag 啥的，看了一下是 dumi 把 `markdown` 解析成 table 组件了，应该要在 dumi 里面去改了，得找一下劈起老师

**Comment by li-jia-nan** [comment]
> Card 组件里也有一个 tabProps，看看需不需要改。

Card 组件的 tabProps 没用到 indicator 属性，应该不用改

File: components/tabs/index.en-US.md (hunk 1 above)
Comment (MadCcc) [reply components/tabs/index.en-US.md:57]: deprecated 的属性一般就从文档里删掉了

**Review by afc163 — APPROVED** [review]

File: components/tabs/index.en-US.md (hunk 1 above)
Comment (li-jia-nan) [reply components/tabs/index.en-US.md]: > deprecated 的属性一般就从文档里删掉了

done

File: components/tabs/index.tsx [hunk 3]
```diff
@@ -115,7 +123,8 @@ const Tabs: React.FC<TabsProps> & { TabPane: typeof TabPane } = (props) => {
       moreIcon={moreIcon}
       prefixCls={prefixCls}
       animated={mergedAnimated}
-      indicatorSize={indicatorSize ?? tabs?.indicatorSize}
+      indicatorSize={legacyIndicatorSize || tabs?.indicatorSize}
+      indicator={{ ...indicator, ...tabs?.indicator }}
```
Comment (MadCcc) [inline components/tabs/index.tsx:127]: 这里就不要用 indicatorSize 了，统一收到 indicator 里

Comment (li-jia-nan) [reply components/tabs/index.tsx:127]: 因为兼容废弃 api 的逻辑在 rc 里面，如果收进来的话，rc 里兼容的逻辑是不是可以删掉了

Comment (afc163) [reply components/tabs/index.tsx:127]: 可以的

**Comment by li-jia-nan** [comment]
先转 draft，我晚点再弄一下

File: components/config-provider/index.en-US.md [hunk 4]
```diff
@@ -147,7 +147,7 @@ const {
 | statistic | Set Statistic common props | { className?: string, style?: React.CSSProperties } | - | 5.7.0 |
 | steps | Set Steps common props | { className?: string, style?: React.CSSProperties } | - | 5.7.0 |
 | table | Set Table common props | { className?: string, style?: React.CSSProperties } | - | 5.7.0 |
-| tabs | Set Tabs common props | { className?: string, style?: React.CSSProperties } | - | 5.7.0 |
+| tabs | Set Tabs common props | { className?: string, style?: React.CSSProperties, size?: GetIndicatorSize, align?: `start` \| `center` \| `end` } | - | 5.7.0 |
```
Comment (MadCcc) [inline components/config-provider/index.en-US.md:150]:
```suggestion
| tabs | Set Tabs common props | { className?: string, style?: React.CSSProperties, indicator?: { size?: GetIndicatorSize, align?: `start` \| `center` \| `end` }} | - | 5.7.0 |
```

File: components/config-provider/index.zh-CN.md [hunk 5]
```diff
@@ -149,7 +149,7 @@ const {
 | statistic | 设置 Statistic 组件的通用属性 | { className?: string, style?: React.CSSProperties } | - | 5.7.0 |
 | steps | 设置 Steps 组件的通用属性 | { className?: string, style?: React.CSSProperties } | - | 5.7.0 |
 | table | 设置 Table 组件的通用属性 | { className?: string, style?: React.CSSProperties } | - | 5.7.0 |
-| tabs | 设置 Tabs 组件的通用属性 | { className?: string, style?: React.CSSProperties } | - | 5.7.0 |
+| tabs | 设置 Tabs 组件的通用属性 | { className?: string, style?: React.CSSProperties, size?: GetIndicatorSize, align?: `start` \| `center` \| `end` } | - | 5.7.0 |
```
Comment (MadCcc) [inline components/config-provider/index.zh-CN.md:152]:
```suggestion
| tabs | 设置 Tabs 组件的通用属性 | { className?: string, style?: React.CSSProperties, indicator?: { size?: GetIndicatorSize, align?: `start` \| `center` \| `end` }} | - | 5.7.0 |
```

File: components/tabs/index.tsx [hunk 6]
```diff
@@ -78,16 +83,27 @@ const Tabs: React.FC<TabsProps> & { TabPane: typeof TabPane } = (props) => {
...
 
   const mergedAnimated = useAnimateConfig(prefixCls, animated);
 
-  const size = useSize(customSize);
-
   const mergedStyle: React.CSSProperties = { ...tabs?.style, ...style };
 
+  const mergedIndicator: TabsProps['indicator'] = {
+    align: indicator?.align || tabs?.indicator?.align,
+    size: indicator?.size || indicatorSize || tabs?.indicator?.size || tabs?.indicatorSize,
```
Comment (MadCcc) [inline components/tabs/index.tsx:104]:
```suggestion
    size: indicator?.size ?? indicatorSize ?? tabs?.indicator?.size ?? tabs?.indicatorSize,
```

File: components/tabs/index.tsx [hunk 7]
```diff
@@ -78,16 +83,27 @@ const Tabs: React.FC<TabsProps> & { TabPane: typeof TabPane } = (props) => {
...
   const mergedItems = useLegacyItems(items, children);
 
   const mergedAnimated = useAnimateConfig(prefixCls, animated);
 
-  const size = useSize(customSize);
-
   const mergedStyle: React.CSSProperties = { ...tabs?.style, ...style };
 
+  const mergedIndicator: TabsProps['indicator'] = {
+    align: indicator?.align || tabs?.indicator?.align,
```
Comment (li-jia-nan) [inline components/tabs/index.tsx:103]:
```suggestion
    align: indicator?.align ?? tabs?.indicator?.align,
```

File: components/tabs/index.zh-CN.md [hunk 8]
```diff
@@ -55,8 +55,7 @@ Ant Design 依次提供了三级选项卡，分别用于不同的场景。
 | centered | 标签居中展示 | boolean | false | 4.4.0 |
 | defaultActiveKey | 初始化选中面板的 key，如果没有设置 activeKey | string | `第一个面板` |  |
 | hideAdd | 是否隐藏加号图标，在 `type="editable-card"` 时有效 | boolean | false |  |
-| indicatorAlign | 自定义指示条对齐方式 | `start` \| `center` \| `end` | `center` | 5.13.0 |
-| indicatorSize | 自定义指示条长度，默认与 tab 等宽 | number \| (origin: number) => number | - | 5.9.0 |
+| indicator | 自定义指示条`长度`和`对齐方式`，默认与 tab 等宽，居中对齐 | { size?: number \| (origin: number) => number; align: `start` \| `center` \| `end`; } | - | 5.13.0 |
```
Comment (MadCcc) [inline components/tabs/index.zh-CN.md:58]:
```suggestion
| indicator | 自定义指示条的长度和对齐方式，默认与 tab 等宽，居中对齐 | { size?: number \| (origin: number) => number; align: `start` \| `center` \| `end`; } | - | 5.13.0 |
```

File: components/tabs/index.en-US.md [hunk 9]
```diff
@@ -53,8 +53,7 @@ Common props ref：[Common props](/docs/react/common-props)
 | centered | Centers tabs | boolean | false | 4.4.0 |
 | defaultActiveKey | Initial active TabPane's key, if `activeKey` is not set | string | - |  |
 | hideAdd | Hide plus icon or not. Only works while `type="editable-card"` | boolean | false |  |
-| indicatorAlign | Customize align of indicator | `start` \| `center` \| `end` | `center` | 5.13.0 |
-| indicatorSize | Customize length of indicator, which is the same as tab by default | number \| (origin: number) => number | - | 5.9.0 |
+| indicator | Customize `length` and `align` of indicator, which is the same as tab by default | { size?: number \| (origin: number) => number; align: `start` \| `center` \| `end`; } | - | 5.13.0 |
```
Comment (MadCcc) [inline components/tabs/index.en-US.md:56]:
```suggestion
| indicator | Customize `size` and `align` of indicator | { size?: number \| (origin: number) => number; align: `start` \| `center` \| `end`; } | - | 5.13.0 |
```

File: components/tabs/index.zh-CN.md [hunk 10]
```diff
@@ -55,8 +55,7 @@ Ant Design 依次提供了三级选项卡，分别用于不同的场景。
 | centered | 标签居中展示 | boolean | false | 4.4.0 |
 | defaultActiveKey | 初始化选中面板的 key，如果没有设置 activeKey | string | `第一个面板` |  |
 | hideAdd | 是否隐藏加号图标，在 `type="editable-card"` 时有效 | boolean | false |  |
-| indicatorAlign | 自定义指示条对齐方式 | `start` \| `center` \| `end` | `center` | 5.13.0 |
-| indicatorSize | 自定义指示条长度，默认与 tab 等宽 | number \| (origin: number) => number | - | 5.9.0 |
+| indicator | 自定义指示条的长度和对齐方式，默认与 tab 等宽，居中对齐 | { size?: number \| (origin: number) => number; align: `start` \| `center` \| `end`; } | - | 5.13.0 |
```
Comment (MadCcc) [inline components/tabs/index.zh-CN.md:58]:
```suggestion
| indicator | 自定义指示条的长度和对齐方式 | { size?: number \| (origin: number) => number; align: `start` \| `center` \| `end`; } | - | 5.13.0 |
```

**Review by MadCcc — APPROVED** [review]

**Pull request merged**